test/integration/auction.test.js

```diff
@@ -62,7 +85,152 @@ describe('Auctions', function () {
+        .get('/auctions/invalidId')
+        .end((err, res) => {
+          if (err) { return done(err) }
+
+          expect(res).to.have.status(404)
+          expect(res.body).to.be.a('object')
+          expect(res.body.message).to.equal('Auction doesn\'t exist')
+
+          return done()
+        })
+    })
+  })
+
+  describe('POST /auctions', function () {
+    it('Should create a new auction', function (done) {
+      chai
+        .request(app)
+        .post('/auctions')
+        .set('cookie', `${cookieName}=${jwt}`)
+        .send(auctionData)
+        .end((err, res) => {
+          if (err) { return done(err) }
+
+          expect(res).to.have.status(201)
+          expect(res.body).to.be.a('object')
+          expect(res.body.message).to.be.equal('Auction created successfully!')
+
+          return done()
+        })
+    })
+
+    it('Should have enough Item Type in wallet', function (done) {
+      chai
+        .request(app)
+        .post('/auctions')
+        .set('cookie', `${cookieName}=${jwt}`)
+        .send({ ...auctionData, quantity: 5 })
+        .end((err, res) => {
+          if (err) { return done(err) }
+
+          expect(res).to.have.status(403)
+          expect(res.body).to.be.a('object')
+          expect(res.body.message).to.be.equal(`You do not have enough of ${itemType}s!`)
+
+          return done()
+        })
+    })
+
+    it('Should return 404, for Auction not Found', function (done) {
+      chai
+        .request(app)
+        .post('/auction')
+        .end((err, res) => {
+          if (err) { return done(err) }
+
+          expect(res).to.have.status(404)
+          expect(res.body).to.be.a('object')
+          expect(res.body).to.deep.equal({
+            statusCode: 404,
+            error: 'Not Found',
+            message: 'Not Found'
+          })
+
+          return done()
+        })
+    })
+  })
+
+  describe('POST /auctions/bid/:id', function () {
+    it('Should make a new bid with given auctionId', function (done) {
+      chai
+        .request(app)
+        .post(`/auctions/bid/${auctionId}`)
+        .set('cookie', `${cookieName}=${jwt}`)
+        .send({ bid: 500 })
+        .end((err, res) => {
+          if (err) { return done(err) }
+
+          expect(res).to.have.status(201)
+          expect(res.body).to.be.a('object')
+          expect(res.body.message).to.be.equal('Successfully placed bid!')
+
+          return done()
+        })
+    })
+
+    it('Should be higher than the previous bid', function (done) {
+      chai
+        .request(app)
+        .post(`/auctions/bid/${auctionId}`)
+        .set('cookie', `${cookieName}=${jwt}`)
+        .send({ bid: 50 })
+        .end((err, res) => {
+          if (err) { return done(err) }
+
+          expect(res).to.have.status(403)
+          expect(res.body).to.be.a('object')
+          expect(res.body.message).to.be.equal('Your bid was not higher than current one!')
+
+          return done()
+        })
+    })
```

Comment:
Can we also have a test for the case when the the user doesn't have sufficient balance?

Reply:
OK, Will be done.

Reply:
What should be the expected minimum balance ?